File: domain-driven-design/go-web/internal/domain/order/order_repository.go
// 领域层(Domain)：订单数据仓库接口
package order

import "go-web-order/internal/domain/repository"

// OrderRepository 订单仓储接口，定义对订单数据的操作。
// type OrderRepository interface {
//   Save(order *Order) error                                  // 保存订单
//   FindByID(id int64) (*Order, error)                        // 根据ID查找订单
//   FindAll(userId int) ([]*Order, error)                     // 查找所有订单
//   Delete(id int64) error                                    // 删除订单
//   FindByCustomerName(customerName string) ([]*Order, error) // 根据客户名称查找订单
// }

// [可选]OrderRepository 订单仓储接口，继承通用仓储接口 Repository[Order]
type OrderRepository interface {
  repository.Repository[Order]  // 继承通用仓储接口
  FindByCustomerName(customerName string) ([]*Order, error) // 根据客户名称查找订单
}

File: domain-driven-design/node-web/src/domain/order/order-repository.js
// src/domain/order/order-repository.js
// 模拟接口供基础层实现，JS因为没有接口，此处用class表示，如果未实现对应函数则发出警告
export default class OrderRepository {
  // 保存实体
  async save(entity) {
    throw new Error('save 方法未实现');
  }

  // 根据ID查找实体
  async findByID(id) {
    throw new Error('findByID 方法未实现');
  }

  // 查找所有实体
  async findAll() {
    throw new Error('findAll 方法未实现');
  }

  // 删除实体
  async delete(id) {
    throw new Error('delete 方法未实现');
  }

  // 根据客户名称查找订单
  async findByCustomerName(customerName) {
    throw new Error('findByCustomerName 方法未实现');
  }
}